src/index.ts

```diff
@@ -509,17 +487,98 @@ const initialize = (apiKey: string, target: Target, options?: Options): Result =
       }
     }
 
-    eventSource.addEventListener('*', (msg: any) => {
-      const event: StreamEvent = JSON.parse(msg.data)
+    const processData = (data: any): void => {
+      const lines = data.toString().split(/\r?\n/);
+      lines.forEach((line) => processLine(line));
```

Comment:
Some opportunities to simplify here
```suggestion
      data.toString().split(/\r?\n/).forEach(processLine);
```